### ISSUE TITLE
Lot's of 404 errors seemingly for shard-number > 0 on testnet

### ISSUE THREAD

**Issue by felixniemeyer** [near/near-lake-framework-js#14]
Hi there, 
I mainly copied the NFT example code when starting to build an indexer for my project. 
I switched to testnet like this
```
const lakeConfig: types.LakeConfig = {
  s3BucketName: "near-lake-data-" + "testnet", // or mainnet
  s3RegionName: "eu-central-1",
  startBlockHeight: 103527488,
};
```
and I am receiving a lot of 404 errors among successful requests like this: 
```

We caught freshly minted NFTs!
[
  {
    receiptId: '2QoYLRBgm6LkH5wzsRqgatadtT8cy5bUbjxv2yXtgEhW',
    marketplace: 'Unknown',
    createdOn: 2022-10-26T11:17:40.663Z,
    nfts: [ { owner_id: 'frenny.testnet', token_ids: [ '#10797' ] } ]
  }
]
Failed to fetch 103527736/shard_3.json. Retrying immediately NoSuchKey: The specified key does not exist.
    at deserializeAws_restXmlNoSuchKeyResponse (/home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/client-s3/dist-cjs/protocols/Aws_restXml.js:6155:23)
    at deserializeAws_restXmlGetObjectCommandError (/home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/client-s3/dist-cjs/protocols/Aws_restXml.js:4353:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24
    at async /home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/middleware-signing/dist-cjs/middleware.js:14:20
    at async StandardRetryStrategy.retry (/home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/middleware-retry/dist-cjs/StandardRetryStrategy.js:51:46)
    at async /home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/middleware-flexible-checksums/dist-cjs/flexibleChecksumsMiddleware.js:56:20
    at async /home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:6:22 {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 404,
    requestId: '8J8DT4B09PCBKPPY',
    extendedRequestId: 'lLu8VxrXExpETA5tBlPe62ZfxNOSy/PnNXlffsW8xGTIyFpOa9qwRE4TOiTM/oQBpY0VFATG5r7JlPqWe4+ChA==',
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  },
  Code: 'NoSuchKey',
  Key: '000103527736/shard_3.json',
  RequestId: '8J8DT4B09PCBKPPY',
  HostId: 'lLu8VxrXExpETA5tBlPe62ZfxNOSy/PnNXlffsW8xGTIyFpOa9qwRE4TOiTM/oQBpY0VFATG5r7JlPqWe4+ChA=='
}
```
I think that maybe testnet has only 1 shard, but there is no number of shards to be configured in the LakeConfig
```
export interface LakeConfig {
    s3Endpoint?: string;
    s3BucketName: string;
    s3RegionName: string;
    startBlockHeight: number;
    blocksPreloadPoolSize?: number;
}
```
How can I use near-lake-framework on testnet without these 404 errors? 
Thanks

**Comment by felixniemeyer** [comment]
Oh, 

the same errors are there on mainnet. E.g. 
```
We caught freshly minted NFTs!
[
  {
    receiptId: 'DN1h3qFVhFU5pHFrRZcYJ3PVL8oNFCmwqNx9V19FWefo',
    marketplace: 'Paras',
    createdOn: 2022-10-26T11:44:17.569Z,
    nfts: [
      {
        owner: 'frnwtr.near',
        links: [ 'https://paras.id/token/x.paras.near::459551/459551:1' ]
      }
    ]
  }
]
Failed to fetch 77069692/shard_3.json. Retrying immediately NoSuchKey: The specified key does not exist.
    at deserializeAws_restXmlNoSuchKeyResponse (/home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/client-s3/dist-cjs/protocols/Aws_restXml.js:6155:23)
    at deserializeAws_restXmlGetObjectCommandError (/home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/client-s3/dist-cjs/protocols/Aws_restXml.js:4353:25)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24
    at async /home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/middleware-signing/dist-cjs/middleware.js:14:20
    at async StandardRetryStrategy.retry (/home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/middleware-retry/dist-cjs/StandardRetryStrategy.js:51:46)
    at async /home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/middleware-flexible-checksums/dist-cjs/flexibleChecksumsMiddleware.js:56:20
    at async /home/felix/dev/web3/chershare/indexer/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:6:22 {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 404,
    requestId: '4DCDHFEY3N2ABVWF',
    extendedRequestId: '5h95FElsbUAJc9tUsRjiAQWWUh2i0Bmu2Cz/K+2aUShM+Kz5R6w+/C3zYYDrIeIZvCR167FFBWw=',
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  },
  Code: 'NoSuchKey',
  Key: '000077069692/shard_3.json',
  RequestId: '4DCDHFEY3N2ABVWF',
  HostId: '5h95FElsbUAJc9tUsRjiAQWWUh2i0Bmu2Cz/K+2aUShM+Kz5R6w+/C3zYYDrIeIZvCR167FFBWw='
}
```
Maybe everything is fine like this and it's normal that there are often no ..._3.json files available for a block? 

In this case, maybe catch the error and inform like this `info: no file for block x in shard available`

Am I doing something wrong?

**Comment by khorolets** [comment]
Hey @felixniemeyer !
Could you clarify whether it is happening while you try to follow the tip of the network? With the latest blocks?

**Comment by khorolets** [comment]
In this case, it's a bit "weird" design on the Framework side. It tries to fetch the `shard` JSON file faster than the Lake Indexer writes. 

I guess we need to silence that "error" message and change it to a warning instead of an error. You might securely ignore it on your side. Also, feel free to open a PR that removes that message.

Ideally, we should:
- wait a bit longer before the next attempt to fetch the file
- show the message when we failed to fetch the file from the second attempt (this might be a sign of the error on the Lake Indexer side)

**Comment by khorolets** [comment]
Closed by #17 Thank you @Abbaskt!
